Certain pasted content freezes multiline input

Reproduce: try and paste this exactly:

```


=================================== FAILURES ===================================
_______________________________ test_micro_help ________________________________
[gw1] linux -- Python 3.11.13 /opt/hostedtoolcache/Python/3.11.13/x64/bin/python

    def test_micro_help():
        """Test that micro --help works correctly."""
        result = subprocess.run(
            [sys.executable, "-m", "microsweagent", "--help"],
            capture_output=True,
            text=True,
            timeout=10,
        )
    
        assert result.returncode == 0
        assert "Run micro-SWE-agent right here, right now." in result.stdout
>       assert "--help" in result.stdout
E       assert '--help' in "👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m ...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n"
E        +  where "👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m ...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n" = CompletedProcess(args=['/opt/hostedtoolcache/Python/3.11.13/x64/bin/python', '-m', 'microsweagent', '--help'], returncode=0, stdout="👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│...le \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n", stderr='').stdout

tests/run/test_cli_integration.py:294: AssertionError
______________________ test_micro_help_with_typer_runner _______________________
[gw1] linux -- Python 3.11.13 /opt/hostedtoolcache/Python/3.11.13/x64/bin/python

    def test_micro_help_with_typer_runner():
        """Test help functionality using typer's test runner."""
        from typer.testing import CliRunner
    
        runner = CliRunner()
        result = runner.invoke(app, ["--help"])
    
        assert result.exit_code == 0
        assert "Run micro-SWE-agent right here, right now." in result.stdout
>       assert "--help" in result.stdout
E       AssertionError: assert '--help' in '\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mmain [OPTIONS]\x1b[0m\x1b[1m                                                         \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m/home/runner/work/micro-swe-agent/micro…\x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n'
E        +  where '\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mmain [OPTIONS]\x1b[0m\x1b[1m                                                         \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m/home/runner/work/micro-swe-agent/micro…\x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n' = <Result okay>.stdout

tests/run/test_cli_integration.py:312: AssertionError
================================ tests coverage ================================
_______________ coverage: platform linux, python 3.11.13-final-0 _______________

Coverage XML written to file coverage.xml
=========================== short test summary info ============================
FAILED tests/run/test_cli_integration.py::test_micro_help - assert '--help' in "👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m ...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n"
 +  where "👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m ...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n" = CompletedProcess(args=['/opt/hostedtoolcache/Python/3.11.13/x64/bin/python', '-m', 'microsweagent', '--help'], returncode=0, stdout="👋 This is micro-swe-agent version 1.0.0.dev3.\nYour config is stored in '/home/runner/.config/micro-swe-agent/.env'\n\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mpython \x1b[0m\x1b[1;32m-m\x1b[0m\x1b[1m microsweagent [OPTIONS]\x1b[0m\x1b[1m                                      \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│...le \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n", stderr='').stdout👋 This is micro-swe-agent version 1.0.0.dev3.
Your config is stored in '/home/runner/.config/micro-swe-agent/.env'

FAILED tests/run/test_cli_integration.py::test_micro_help_with_typer_runner - AssertionError: assert '--help' in '\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mmain [OPTIONS]\x1b[0m\x1b[1m                                                         \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m/home/runner/work/micro-swe-agent/micro…\x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n'
 +  where '\x1b[1m                                                                                \x1b[0m\n\x1b[1m \x1b[0m\x1b[1;33mUsage: \x1b[0m\x1b[1mmain [OPTIONS]\x1b[0m\x1b[1m                                                         \x1b[0m\x1b[1m \x1b[0m\n\x1b[1m                                                                                \x1b[0m\n Run micro-SWE-agent right here, right now.                                     \n                                                                                \n                                                                                \n\x1b[2m╭─\x1b[0m\x1b[2m Options \x1b[0m\x1b[2m───────────────────────────────────────────────────────────────────\x1b[0m\x1b[2m─╮\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-config\x1b[0m              \x1b[1;32m-c\x1b[0m      \x1b[1;33mPATH\x1b[0m  Path to config file                      \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m[default:                               \x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     \x1b[2m/home/runner/work/micro-swe-agent/micro…\x1b[0m \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1...m  Output file \x1b[2m[default: None]\x1b[0m              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-visual\x1b[0m              \x1b[1;32m-v\x1b[0m      \x1b[1;33m    \x1b[0m  Use visual UI (Textual)                  \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-install\x1b[0m\x1b[1;36m-completion\x1b[0m          \x1b[1;33m    \x1b[0m  Install completion for the current       \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     shell.                                   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-show\x1b[0m\x1b[1;36m-completion\x1b[0m             \x1b[1;33m    \x1b[0m  Show completion for the current shell,   \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     to copy it or customize the              \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m                                     installation.                            \x1b[2m│\x1b[0m\n\x1b[2m│\x1b[0m \x1b[1;36m-\x1b[0m\x1b[1;36m-help\x1b[0m                        \x1b[1;33m    \x1b[0m  Show this message and exit.              \x1b[2m│\x1b[0m\n\x1b[2m╰──────────────────────────────────────────────────────────────────────────────╯\x1b[0m\n\n' = <Result okay>.stdout
================== 2 failed, 208 passed in 200.04s (0:03:20) ===================
Error: Process completed with exit code 1.
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,6 +40,7 @@ dependencies = [
     "typer",
     "platformdirs",
     "textual",
+    "prompt_toolkit",
 ]
 
 [project.optional-dependencies]

--- a/src/microsweagent/__init__.py
+++ b/src/microsweagent/__init__.py
@@ -20,9 +20,9 @@ from rich.console import Console
 
 package_dir = Path(__file__).resolve().parent
 
-config_dir = Path(os.getenv("MICRO_SWE_AGENT_GLOBAL_CONFIG_DIR") or user_config_dir("micro-swe-agent"))
-config_dir.mkdir(parents=True, exist_ok=True)
-global_config_file = Path(config_dir) / ".env"
+global_config_dir = Path(os.getenv("MICRO_SWE_AGENT_GLOBAL_CONFIG_DIR") or user_config_dir("micro-swe-agent"))
+global_config_dir.mkdir(parents=True, exist_ok=True)
+global_config_file = Path(global_config_dir) / ".env"
 
 if not os.getenv("MICRO_SWE_AGENT_SILENT_STARTUP"):
     Console().print(
@@ -64,4 +64,4 @@ class Agent(Protocol):
     def run(self, task: str) -> tuple[str, str]: ...
 
 
-__all__ = ["Agent", "Model", "Environment", "package_dir", "__version__", "global_config_file"]
+__all__ = ["Agent", "Model", "Environment", "package_dir", "__version__", "global_config_file", "global_config_dir"]

--- a/src/microsweagent/agents/interactive.py
+++ b/src/microsweagent/agents/interactive.py
@@ -10,12 +10,16 @@ import re
 from dataclasses import dataclass, field
 from typing import Literal
 
+from prompt_toolkit.history import FileHistory
+from prompt_toolkit.shortcuts import PromptSession
 from rich.console import Console
 from rich.rule import Rule
 
+from microsweagent import global_config_dir
 from microsweagent.agents.default import AgentConfig, DefaultAgent, NonTerminatingException
 
 console = Console(highlight=False)
+prompt_session = PromptSession(history=FileHistory(global_config_dir / "interactive_history.txt"))
 
 
 @dataclass
@@ -65,7 +69,7 @@ class InteractiveAgent(DefaultAgent):
             # We always add a message about the interrupt and then just proceed to the next step
             interruption_message = self._prompt_and_handle_special(
                 "\n\n[bold yellow]Interrupted.[/bold yellow] "
-                "[green]type a comment/command[/green] (/h for available commands)"
+                "[green]Type a comment/command[/green] (/h for available commands)"
                 "\n[bold yellow]>[/bold yellow] "
             ).strip()
             if not interruption_message or interruption_message in self._MODE_COMMANDS_MAPPING:
@@ -84,7 +88,7 @@ class InteractiveAgent(DefaultAgent):
     def ask_confirmation(self) -> None:
         prompt = (
             "[bold yellow]Execute?[/bold yellow] [green][bold]Enter[/bold] to confirm[/green], "
-            "or [green]type a comment/command[/green] (/h for available commands)\n"
+            "or [green]Type a comment/command[/green] (/h for available commands)\n"
             "[bold yellow]>[/bold yellow] "
         )
         match user_input := self._prompt_and_handle_special(prompt).strip():
@@ -99,7 +103,8 @@ class InteractiveAgent(DefaultAgent):
 
     def _prompt_and_handle_special(self, prompt: str) -> str:
         """Prompts the user, takes care of /h (followed by requery) and sets the mode. Returns the user input."""
-        user_input = console.input(prompt).strip()
+        console.print(prompt, end="")
+        user_input = prompt_session.prompt("")
         if user_input == "/h":
             console.print(
                 f"Current mode: [bold green]{self.config.mode}[/bold green]\n"

--- a/src/microsweagent/run/micro.py
+++ b/src/microsweagent/run/micro.py
@@ -6,9 +6,11 @@ from typing import Any
 
 import typer
 import yaml
+from prompt_toolkit.history import FileHistory
+from prompt_toolkit.shortcuts import PromptSession
 from rich.console import Console
 
-from microsweagent import Environment, Model
+from microsweagent import Environment, Model, global_config_dir
 from microsweagent.agents.interactive import InteractiveAgent
 from microsweagent.agents.interactive_textual import AgentApp
 from microsweagent.config import builtin_config_dir, get_config_path
@@ -19,22 +21,7 @@ from microsweagent.run.utils.save import save_traj
 DEFAULT_CONFIG = Path(os.getenv("MSWEA_LOCAL_CONFIG_PATH", builtin_config_dir / "local.yaml"))
 console = Console(highlight=False)
 app = typer.Typer()
-
-
-def get_multiline_task() -> str:
-    """Get a multiline task/problem statement from the user."""
-    console.print(
-        "[bold yellow]What do you want to do?[/bold yellow] (press [bold red]Ctrl+D[/bold red] in a [red]new line[/red] to finish)"
-    )
-    lines = []
-    while True:
-        try:
-            lines.append(console.input("[bold green]>[/bold green] "))
-        except EOFError:
-            break
-
-    print()
-    return "\n".join(lines).strip()
+prompt_session = PromptSession(history=FileHistory(global_config_dir / "micro_task_history.txt"))
 
 
 def run_interactive(model: Model, env: Environment, agent_config: dict, task: str, output: Path | None = None) -> Any:
@@ -88,7 +75,15 @@ def main(
     _config = yaml.safe_load(get_config_path(config).read_text())
 
     if not task:
-        task = get_multiline_task()
+        console.print(
+            "[bold yellow]What do you want to do?\n"
+            "[bold green]Up[/bold green]/[bold green]Down[/bold green] to bring up previous tasks or [bold green]Ctrl+R[/bold green] to search\n"
+            "Confirm input with [bold green]Esc, Enter[/bold green]\n"
+            ">[/bold yellow] ",
+            end="",
+        )
+        task = prompt_session.prompt("", multiline=True)
+        console.print("[bold green]Got that, thanks![/bold green]")
 
     mode = "confirm" if not yolo else "yolo"
 
@@ -97,9 +92,9 @@ def main(
     agent_config = _config.get("agent", {}) | {"mode": mode}
 
     if visual:
-        return run_textual(_model, env, agent_config, task, output)
+        return run_textual(_model, env, agent_config, task, output)  # type: ignore[arg-type]
     else:
-        return run_interactive(_model, env, agent_config, task, output)
+        return run_interactive(_model, env, agent_config, task, output)  # type: ignore[arg-type]
 
 
 if __name__ == "__main__":

--- a/tests/agents/test_interactive.py
+++ b/tests/agents/test_interactive.py
@@ -7,7 +7,7 @@ from microsweagent.models.test_models import DeterministicModel
 
 def test_successful_completion_with_confirmation():
     """Test agent completes successfully when user confirms all actions."""
-    with patch("microsweagent.agents.interactive.console.input", side_effect=[""]):  # Confirm action with Enter
+    with patch("microsweagent.agents.interactive.prompt_session.prompt", side_effect=[""]):  # Confirm action with Enter
         agent = InteractiveAgent(
             model=DeterministicModel(
                 outputs=["Finishing\n```bash\necho 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'completed'\n```"]
@@ -24,7 +24,7 @@ def test_successful_completion_with_confirmation():
 def test_action_rejection_and_recovery():
     """Test agent handles action rejection and can recover."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "User rejected this action",  # Reject first action
             "",  # Confirm second action
@@ -52,7 +52,7 @@ def test_action_rejection_and_recovery():
 def test_yolo_mode_activation():
     """Test entering yolo mode disables confirmations."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/y",  # Enter yolo mode
             "",  # This should be ignored since yolo mode is on
@@ -74,7 +74,7 @@ def test_yolo_mode_activation():
 def test_help_command():
     """Test help command shows help and continues normally."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/h",  # Show help
             "",  # Confirm action after help
@@ -145,7 +145,7 @@ def _test_interruption_helper(interruption_input, expected_message_fragment, pro
             return interruption_input  # For the interruption handling
         return ""  # Confirm all subsequent actions
 
-    with patch("microsweagent.agents.interactive.console.input", side_effect=mock_input):
+    with patch("microsweagent.agents.interactive.prompt_session.prompt", side_effect=mock_input):
         with patch.object(agent, "query", side_effect=mock_query):
             exit_status, result = agent.run(problem_statement)
 
@@ -174,7 +174,7 @@ def test_interruption_handling_empty_message():
 def test_multiple_confirmations_and_commands():
     """Test complex interaction with multiple confirmations and commands."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "reject first",  # Reject first action
             "/h",  # Show help for second action
@@ -201,7 +201,7 @@ def test_multiple_confirmations_and_commands():
 
 def test_non_whitelisted_action_requires_confirmation():
     """Test that non-whitelisted actions still require confirmation."""
-    with patch("microsweagent.agents.interactive.console.input", return_value=""):
+    with patch("microsweagent.agents.interactive.prompt_session.prompt", return_value=""):
         agent = InteractiveAgent(
             model=DeterministicModel(
                 outputs=["Non-whitelisted\n```bash\necho 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'confirmed'\n```"]
@@ -221,7 +221,7 @@ def test_non_whitelisted_action_requires_confirmation():
 def test_human_mode_basic_functionality():
     """Test human mode where user enters shell commands directly."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "echo 'user command'",  # User enters shell command
             "echo 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'human mode works'",  # User enters final command
@@ -243,7 +243,7 @@ def test_human_mode_basic_functionality():
 def test_human_mode_switch_to_yolo():
     """Test switching from human mode to yolo mode."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/y",  # Switch to yolo mode from human mode
             "",  # Confirm action in yolo mode (though no confirmation needed)
@@ -267,7 +267,7 @@ def test_human_mode_switch_to_yolo():
 def test_human_mode_switch_to_confirm():
     """Test switching from human mode to confirm mode."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/c",  # Switch to confirm mode from human mode
             "",  # Confirm action in confirm mode
@@ -291,7 +291,7 @@ def test_human_mode_switch_to_confirm():
 def test_confirmation_mode_switch_to_human_with_rejection():
     """Test switching from confirm mode to human mode with /u command."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/u",  # Switch to human mode and reject action
             "echo 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'human command after rejection'",  # Human command
@@ -320,7 +320,7 @@ def test_confirmation_mode_switch_to_human_with_rejection():
 def test_confirmation_mode_switch_to_yolo_and_continue():
     """Test switching from confirm mode to yolo mode with /y and continuing with action."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/y",  # Switch to yolo mode and confirm current action
         ],
@@ -364,7 +364,7 @@ def test_mode_switch_during_keyboard_interrupt():
         return original_query(*args, **kwargs)
 
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/y",  # Switch to yolo mode during interrupt
             "",  # Confirm subsequent actions (though yolo mode won't ask)
@@ -384,7 +384,7 @@ def test_mode_switch_during_keyboard_interrupt():
 def test_already_in_mode_behavior():
     """Test behavior when trying to switch to the same mode."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/c",  # Try to switch to confirm mode when already in confirm mode
             "",  # Confirm action after the "already in mode" recursive prompt
@@ -407,7 +407,7 @@ def test_already_in_mode_behavior():
 def test_all_mode_transitions_yolo_to_others():
     """Test transitions from yolo mode to other modes."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/c",  # Switch from yolo to confirm
             "",  # Confirm action in confirm mode
@@ -447,7 +447,7 @@ def test_all_mode_transitions_yolo_to_others():
 def test_all_mode_transitions_confirm_to_human():
     """Test transition from confirm mode to human mode."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/u",  # Switch from confirm to human (rejecting action)
             "echo 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'human command'",  # User enters command in human mode
@@ -469,7 +469,7 @@ def test_help_command_from_different_contexts():
     """Test help command works from different contexts (confirmation, interrupt, human mode)."""
     # Test help during confirmation
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/h",  # Show help during confirmation
             "",  # Confirm after help
@@ -495,7 +495,7 @@ def test_help_command_from_different_contexts():
 def test_help_command_from_human_mode():
     """Test help command works from human mode."""
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/h",  # Show help in human mode
             "echo 'MICRO_SWE_AGENT_FINAL_OUTPUT'\necho 'help in human mode'",  # User command after help
@@ -542,7 +542,7 @@ def test_complex_mode_switching_sequence():
         return original_query(*args, **kwargs)
 
     with patch(
-        "microsweagent.agents.interactive.console.input",
+        "microsweagent.agents.interactive.prompt_session.prompt",
         side_effect=[
             "/y",  # Confirm->Yolo during first action confirmation
             "/u",  # Yolo->Human during interrupt

--- a/tests/run/test_cli_integration.py
+++ b/tests/run/test_cli_integration.py
@@ -34,7 +34,7 @@ def test_micro_command_calls_run_interactive():
         mock_yaml_load.return_value = {"agent": {"system_template": "test"}, "env": {}, "model": {}}
         mock_run_interactive.return_value = Mock()
 
-        # Call main function with task provided (so get_multiline_task is not called)
+        # Call main function with task provided (so prompt is not called)
         main(
             config=DEFAULT_CONFIG,
             model="test-model",
@@ -90,10 +90,10 @@ def test_micro_v_command_calls_run_textual():
         assert args[3] == "Test task"  # task
 
 
-def test_micro_calls_get_multiline_task_when_no_task_provided():
-    """Test that micro calls get_multiline_task when no task is provided."""
+def test_micro_calls_prompt_when_no_task_provided():
+    """Test that micro calls prompt when no task is provided."""
     with (
-        patch("microsweagent.run.micro.get_multiline_task") as mock_get_multiline_task,
+        patch("microsweagent.run.micro.prompt_session.prompt") as mock_prompt,
         patch("microsweagent.run.micro.run_interactive") as mock_run_interactive,
         patch("microsweagent.run.micro.get_model") as mock_get_model,
         patch("microsweagent.run.micro.LocalEnvironment") as mock_env,
@@ -101,7 +101,7 @@ def test_micro_calls_get_multiline_task_when_no_task_provided():
         patch("microsweagent.run.micro.yaml.safe_load") as mock_yaml_load,
     ):
         # Setup mocks
-        mock_get_multiline_task.return_value = "User provided task"
+        mock_prompt.return_value = "User provided task"
         mock_model = Mock()
         mock_get_model.return_value = mock_model
         mock_environment = Mock()
@@ -122,19 +122,19 @@ def test_micro_calls_get_multiline_task_when_no_task_provided():
             visual=False,
         )
 
-        # Verify get_multiline_task was called
-        mock_get_multiline_task.assert_called_once()
+        # Verify prompt was called
+        mock_prompt.assert_called_once()
 
-        # Verify run_interactive was called with the task from get_multiline_task
+        # Verify run_interactive was called with the task from prompt
         mock_run_interactive.assert_called_once()
         args, kwargs = mock_run_interactive.call_args
         assert args[3] == "User provided task"  # task
 
 
-def test_micro_v_calls_get_multiline_task_when_no_task_provided():
-    """Test that micro -v calls get_multiline_task when no task is provided."""
+def test_micro_v_calls_prompt_when_no_task_provided():
+    """Test that micro -v calls prompt when no task is provided."""
     with (
-        patch("microsweagent.run.micro.get_multiline_task") as mock_get_multiline_task,
+        patch("microsweagent.run.micro.prompt_session.prompt") as mock_prompt,
         patch("microsweagent.run.micro.run_textual") as mock_run_textual,
         patch("microsweagent.run.micro.get_model") as mock_get_model,
         patch("microsweagent.run.micro.LocalEnvironment") as mock_env,
@@ -142,7 +142,7 @@ def test_micro_v_calls_get_multiline_task_when_no_task_provided():
         patch("microsweagent.run.micro.yaml.safe_load") as mock_yaml_load,
     ):
         # Setup mocks
-        mock_get_multiline_task.return_value = "User provided visual task"
+        mock_prompt.return_value = "User provided visual task"
         mock_model = Mock()
         mock_get_model.return_value = mock_model
         mock_environment = Mock()
@@ -163,10 +163,10 @@ def test_micro_v_calls_get_multiline_task_when_no_task_provided():
             visual=True,
         )
 
-        # Verify get_multiline_task was called
-        mock_get_multiline_task.assert_called_once()
+        # Verify prompt was called
+        mock_prompt.assert_called_once()
 
-        # Verify run_textual was called with the task from get_multiline_task
+        # Verify run_textual was called with the task from prompt
         mock_run_textual.assert_called_once()
         args, kwargs = mock_run_textual.call_args
         assert args[3] == "User provided visual task"  # task

